webui/react/src/omnibar/Component.tsx

```diff
@@ -0,0 +1,48 @@
+import OmnibarNpm from 'omnibar';
+import React, { useEffect } from 'react';
+import { GlobalHotKeys } from 'react-hotkeys';
+
+import { StoreAction, useStore, useStoreDispatch } from 'contexts/Store';
+import { exposeStore } from 'omnibar/exposedStore';
+import BaseRenderer from 'omnibar/modifiers/anchor/BaseRenderer';
+import * as Tree from 'omnibar/Tree';
+
+import { BaseNode } from './AsyncTree';
+import css from './Component.module.scss';
+
+const globalKeymap = { HIDE_OMNIBAR: [ 'esc' ] };
+
+const Omnibar: React.FC = () => {
+  const storeDispatch = useStoreDispatch();
+  const { omnibar } = useStore();
+  const globalKeyHandler = {
+    HIDE_OMNIBAR: (): void => {
+      storeDispatch({ type: StoreAction.HideOmnibar });
+    },
+    SHOW_OMNIBAR: (): void => storeDispatch({ type: StoreAction.ShowOmnibar }),
+  };
+
+  useEffect(() => {
+    exposeStore({ dispatch: storeDispatch, state: { omnibar } });
```

Comment:
Not the safest thing to do but gives us access to the store without turning different omnibar trees into React components.

Reply:
**suggestion:** this feels like an anti-pattern. How about defining a `const handleAction = useCallback(..., []);` inside `Component.tsx` and in there put all the logic that is in `Tree.onAction`. Taking a look at `Tree.ts` more closely, it seems like it is a good candidate to split up into half into `omnibar/utils` and half directly into `Component.tsx`. Especially `onAction` and `extension` since they are React specific and don't really need to be extracted.